### ISSUE TITLE
Issues with the countries and country names

### ISSUE THREAD

**Issue by ElsaRaunioIFRC** [IFRCGo/go-frontend#415]
Since there are several cards related to the country names, I am trying to compile all of them here.

An up to date list of countries and their respective National Societies (where relevant) can be found here:

https://docs.google.com/spreadsheets/d/1mw0uQ-tJtcucBAhmgro-oHWjKTb5AxHlJe9e2TwMvz8/edit?usp=sharing

Only countries listed in the Main sheet can be presented on the Regional Pages of the Go platform. Similarly, each country should be referred to with the name presented in Column A. Column E shows the region where each country belongs.

**Comment by ElsaRaunioIFRC** [comment]
ElsaRaunioIFRC commented 14 days ago
Serbia and Montenegro ceased to exist in 2006. However, we still have it in the list of countries in the Europe Region page.

I have tried to unassign Serbia and Montenegro from the Europe Region on Django. However, this does not seem to work as the system keeps prompting for a region.

While we want to keep Serbia and Montenegro in our database (because of historical operations in the country), we should not display it for the public in our list of countries.

 @ElsaRaunioIFRC ElsaRaunioIFRC added the Must Fix  label 14 days ago
@dereklieu
  Member
dereklieu commented 14 days ago
This could be accomplished by adding a inactive field to the country model, and then modifying the front-end to filter those that are set to inactive.
 @ElsaRaunioIFRC
  Member
ElsaRaunioIFRC commented 13 days ago
Thanks for the response @dereklieu. Would this work with the same principle as the active/inactive users?
 @dereklieu
  Member
dereklieu commented 13 days ago
Not exactly, active/inactive is a Django feature of the user model. In this case we would have to build our own functionality on the front-end to hide inactive countries.

**Comment by ElsaRaunioIFRC** [comment]
After the discussion with the IFRC Governance, it was confirmed that Israel is part of the Europe Region (and not MENA as is currently the case on GO). Hence, despite the geography, we should move Israel into the list of countries in Europe. This should be the case in the maps as well.

**Comment by dereklieu** [comment]
@ElsaRaunioIFRC We currently keep two copies of country names, on in the database, and one in the Go website application.

To change the value in the database, just use the admin. This will update the website **tables**.

We keep the second version in the Go website to power a few things, namely the field report dropdown form. Ideally in a later phase this should also query the API/database, so that there is just one source for countries. However, in the meantime, we should manually propagate any changes to the website as well. That data is currently located [here](https://github.com/IFRCGo/go-frontend/blob/develop/app/assets/scripts/utils/field-report-constants.js#L32).

Long story short, you can change country names in the admin, just keep a record of what you've changed, and I can make those same changes to the field report dropdown form on go.ifrc.org. The data tables should update as soon as you change them in the admin.

You can also reassign Israel to a new region using the admin. However, we may have to adjust the bounding box of the region to fit. Here are the current [bounding box specs](https://github.com/IFRCGo/go-frontend/blob/develop/app/assets/scripts/utils/region-bounding-box.js). Europe is `3`.

You can also delete zones that no longer exist. Just tell me and I'll delete them in the dropdown form as well.

**Comment by dereklieu** [comment]
Also, I always recommend making these changes in the dev environment first.

For Serbia/Montenegro, I'm not sure we arrived at a conclusion. My recommendation was to add a flag to the country model, which would prevent it from showing up in country lists on regional pages. However, thinking about it more, I could also just remove the need to specify a region. This would accomplish the same thing. @ElsaRaunioIFRC Thoughts?

This would also fix https://github.com/IFRCGo/go-frontend/issues/407 as it would allow you to add IFRC and ICRC as countries, essentially allowing you to display them as deployment sources.

**Comment by ElsaRaunioIFRC** [comment]
@dereklieu As for changing the names of the countries, regions and zones, I can proceed with that today. I will provide you a list of the changes I'll make. How about the country database in the GitHub? Would you prefer to make the edits there or shall our team do them?

**Comment by dereklieu** [comment]
@ElsaRaunioIFRC if you go ahead and change the names in the admin, I'm happy to make the changes in Github, or have your team do it if you feel comfortable doing that - just let me know :)

**Comment by ElsaRaunioIFRC** [comment]
@dereklieu Now that I've been working on the GitHub database, I've noticed that it does not contain names of the countries that do not currently have their own national societies. However, after a brief discussion with the team, we suggest adding those countries in the country lists. Even without their own national societies, these countries can still have partner national societies working in their territory. This is the case, for example, with Eritrea.

The following countries, recognised by the IFRC, do not have recognised national societies:
- Bhutan (the newly launched Red Cross Society will become a member of the IFRC in the near future)
- Eritrea
- Holy See (aka Vatican)
- Nauru
- Niue (I'm trying to get a final confirmation of the status)
- Oman

**Comment by ElsaRaunioIFRC** [comment]
@dereklieu It seems like it is still not possible to add or edit a country without assigning it into a specific region. This would be necessary in order to remove Serbia & Montenegro from the Europe Region page. Also, this feature is essential for adding country cluster offices and country offices into the database.

**Comment by dereklieu** [comment]
@szabozoltan69 can you work on this? It would just mean removing the required property from the region relation in the country model.

**Comment by szabozoltan69** [comment]
Yes, I can do it tomorrow. Z.

Derek Lieu <notifications@github.com> ezt írta (időpont: 2018. szept. 17.,
H 17:19):

**Comment by szabozoltan69** [comment]
@dereklieu – now it is done, please review the https://github.com/IFRCGo/go-api/pull/293 pull request.

**Comment by dereklieu** [comment]
@ElsaRaunioIFRC @szabozoltan69 can this be closed?

**Comment by szabozoltan69** [comment]
Yes, the task is done.

**Comment by nanometrenat** [comment]
xref https://github.com/IFRCGo/go-frontend/issues/415

**Comment by nanometrenat** [comment]
xref https://github.com/IFRCGo/go-frontend/issues/938
